### PR TITLE
Fix wrong time and clean up logbook Recent rows

### PR DESCRIPTION
**Pull request by patrickrb**
## Problem

The Logbook → Recent (Недавние) tab showed the **same wrong time on every row** and a clipped, cramped band column (screenshot: every row read `20:26`, band showed `20M(14.084`).

- **Wrong time:** `formatTime()` sliced the `qso_date` column (`20260611`) characters 0–1 / 2–3 → `20:26`, i.e. the year+month of the *date*, not a time of day. The real `time_on` is already loaded into `QSLCallsignRecord` and sorted on by `dev` — only the display was wrong.
- **Clipped band:** the `band` field is the SQL concatenation `20M(14.084 MHz)`, rendered in a 52dp column with `maxLines=1` and no ellipsis, so it hard-clipped to `20M(14.084`. The packed string also isn't a `BandColorMap` key, so `bandColor()` fell through to muted gray.

## Change

- Replace the broken `formatTime()` with four testable `internal` helpers:
  - `formatQsoTime()` — renders normalized `time_on` as `HH:MM` (reuses existing `normalizeTimeOn`).
  - `formatQsoDate()` — `yyyyMMdd` → short `"11 Jun"`.
  - `parseBandMeter()` / `parseFreqMhz()` — split the packed `"20M(14.084 MHz)"` column.
- Rewrite `QsoRow` to a **band-chip-lead** layout: a color-coded band chip on the left, callsign + grid/country, and a muted `14.084 MHz · 20:26z · 11 Jun` meta line (UTC time + short date). Removes the clipping 52dp column.

Backend is untouched — `time_on` was already wired up on `dev`; this is a display-only fix.

```
┌────────────────────────────────────────┐
│ ╔════╗  UB6C             ✓  ↑QRZ      ⋮ │
│ ║20M ║  KN94 · United States            │
│ ╚════╝  14.084 MHz · 20:26z · 11 Jun    │
└────────────────────────────────────────┘
```

## Tests

New `LogbookRowLogicTest` covers all four helpers, including the dropped-leading-zero time and malformed-date edge cases. Passes alongside the existing `LogbookSortTest`:

```
gradlew.bat testDebugUnitTest --tests *LogbookRowLogicTest --tests *LogbookSortTest  → BUILD SUCCESSFUL
```

## Verification status

APK builds cleanly. On-device visual check is **still pending** — no phone was connected at build time (`installDebug` → "No connected devices!"). Needs a quick look on the Pixel 8: Journal → Недавние should show distinct real UTC times, the colored band chip, and an un-clipped frequency.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ft8cn/app/src/main/kotlin/radio/ks3ckc/ft8us/ui/logbook/LogbookScreen.kt [hunk 1]
```diff
@@ -1349,18 +1374,53 @@ private fun SyncChip(label: String) {
...
+ * Format a stored time-of-day (`time_on`, UTC) as "HH:MM". The SQL query already
+ * normalizes `time_on` to 6-digit HHMMSS, but we route through [normalizeTimeOn]
+ * anyway so odd-width inputs (HHMM, or a dropped leading zero) still render right.
+ * Blank / non-numeric input → "--:--".
+ */
+internal fun formatQsoTime(timeOn: String): String {
+    if (timeOn.none { it.isDigit() }) return "--:--"
+    val norm = normalizeTimeOn(timeOn)
+    return "${norm.substring(0, 2)}:${norm.substring(2, 4)}"
+}
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ft8cn/app/src/main/kotlin/radio/ks3ckc/ft8us/ui/logbook/LogbookScreen.kt:1394]: Fixed in 01aae291. The guard now rejects any input that isn't purely numeric — `if (timeOn.isEmpty() || timeOn.any { !it.isDigit() }) return "--:--"` — instead of only the all-non-digit case. So "12:30" and "12ab" return "--:--" rather than being silently stripped to digits by `normalizeTimeOn` and rendered as a real time, matching the KDoc contract. Added `formatQsoTime("12ab")` and `formatQsoTime("12:30")` → "--:--" to the test.

File: ft8cn/app/src/main/kotlin/radio/ks3ckc/ft8us/ui/logbook/LogbookScreen.kt [hunk 2]
```diff
@@ -1349,18 +1374,53 @@ private fun SyncChip(label: String) {
...
+    if (digits.length < 8) return ""
+    val month = digits.substring(4, 6).toIntOrNull() ?: return ""
+    val day = digits.substring(6, 8).toIntOrNull() ?: return ""
+    if (month !in 1..12 || day !in 1..31) return ""
+    val months = arrayOf(
+        "Jan", "Feb", "Mar", "Apr", "May", "Jun",
+        "Jul", "Aug", "Sep", "Oct", "Nov", "Dec",
+    )
+    return "$day ${months[month - 1]}"
+}
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ft8cn/app/src/main/kotlin/radio/ks3ckc/ft8us/ui/logbook/LogbookScreen.kt:1413]: Fixed in 01aae291. `formatQsoDate` now requires exactly 8 digit characters — `if (qsoDate.length != 8 || qsoDate.any { !it.isDigit() }) return ""` — instead of stripping non-digits and accepting the first 8 of any ≥8-digit string. So "20260611xxx" and "2026-06-11" now return "" per the documented 8-digit yyyyMMdd contract. Added both as test cases.
